flake.nix

```diff
@@ -0,0 +1,49 @@
+{
+  description = "Types and optics for manipulating Amazon Resource Names (ARNs)";
+
+  inputs = {
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+    flake-utils.url = "github:numtide/flake-utils";
+    nixpkgs.url = "github:nixos/nixpkgs";
+  };
+
+  outputs = inputs:
+    inputs.flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import inputs.nixpkgs { inherit system; };
+        makePackage = haskellPackages: (haskellPackages.override {
+          overrides = final: prev: with pkgs.haskell.lib; {
+            hedgehog = prev.callHackage "hedgehog" "1.2" { };
```

Comment:
Why not to use haskell.nix?

Reply:
For external FOSS projects, I prefer to stick with straight nixpkgs if possible, as it avoids having to ask devs to set up their cache correctly. `wai-handler-hal` is a special case because we need to build static musl binaries in the example.

Reply:
I agree to avoid `haskell.nix` if the project is simple.

> as it avoids having to ask devs to set up their cache correctly

By the way, there is a trick to add cache in flake.nix
```nix
  nixConfig = {
    extra-substituters = [ "https://cache.iog.io" ];
    extra-trusted-public-keys =
      [ "hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ=" ];
    allow-import-from-derivation = "true";
  };
```